src/NLog.Web/NLogRequestLoggingModule.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Web;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// HttpModule that writes all requests to Logger named "RequestLogging"
+    /// </summary>
+    public class NLogRequestLoggingModule : IHttpModule
+    {
+        private static readonly NLog.Logger Logger = NLog.LogManager.GetLogger("RequestLogging");
+
+        void IHttpModule.Init(HttpApplication context)
+        {
+            context.EndRequest += LogHttpRequest;
+        }
+
+        private void LogHttpRequest(object sender, EventArgs e)
+        {
+            Exception exception = null;
+            int statusCode = 400;
+
+            try
+            {
+                exception = HttpContext.Current?.Server?.GetLastError();
+                statusCode = HttpContext.Current?.Response?.StatusCode ?? 400;
+            }
+            catch
+            {
+                // Nothing to do
+            }
+            finally
+            {
+                if (exception != null)
+                    Logger.Error(exception, "HttpRequest Exception");
+                else if (statusCode >= 400)
```

Comment:
I guess we could also have  a 500+ without exception, I guess that should be an error

Reply:
Not a ASP.NET developer, so clue when a HttpStatusCode should signal a warning or error.